### PR TITLE
Fixed iOS null event crash issue

### PR DESCRIPTION
**Pull request by rakeshghasadiya**

### CODE REVIEW

**Comment by alindao-charles** [comment]
Please merge this

File: RNCalendarEvents.m [hunk 1]
```diff
@@ -30,19 +30,10 @@ @implementation RNCalendarEvents
...
-    }else{
-        r = 1;
-        g = 1;
-        b = 1;
-    }
-    
+    CGFloat r = components[0];
+    CGFloat g = components[1];
+    CGFloat b = components[2];
+
```
Comment (MoOx) [inline RNCalendarEvents.m:36]: Are you sure about this change?
This seems more defensive programming that could be legit (I didn't check the git history to find out)

File: RNCalendarEvents.m [hunk 2]
```diff
@@ -173,7 +164,7 @@ - (NSDictionary *)buildAndSaveEvent:(NSDictionary *)details options:(NSDictionar
         calendarEvent.availability = [self availablilityConstantMatchingString:availability];
     }
 
-    NSURL *URL = [NSURL URLWithString:[url stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLQueryAllowedCharacterSet]]];
+    NSURL *URL = [NSURL URLWithString:[url stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet URLHostAllowedCharacterSet]]];
```
Comment (MoOx) [inline RNCalendarEvents.m:167]: We recently made a fix somewhere, are you also sure about this change?

File: RNCalendarEvents.m [hunk 3]
```diff
@@ -458,16 +449,20 @@ - (NSArray *)serializeCalendarEvents:(NSArray *)calendarEvents
     NSMutableArray *serializedCalendarEvents = [[NSMutableArray alloc] init];
 
     for (EKEvent *event in calendarEvents) {
-
-        [serializedCalendarEvents addObject:[self serializeCalendarEvent:event]];
+        NSDictionary * sCalendarEvent = [self serializeCalendarEvent:event];
+        if(sCalendarEvent != nil){
+            [serializedCalendarEvents addObject:sCalendarEvent];
+        }
```
Comment (MoOx) [inline RNCalendarEvents.m:455]: This look like it's the fix you want to push?

File: RNCalendarEvents.m [hunk 4]
```diff
@@ -680,7 +675,11 @@ - (NSDictionary *)serializeCalendarEvent:(EKEvent *)event
         [formedCalendarEvent setValue: structuredLocation forKey:@"structuredLocation"];
     }
 
-    return [formedCalendarEvent copy];
+        return [formedCalendarEvent copy];
+    } @catch (NSException *exception) {
+        return nil;
+    }
```
Comment (MoOx) [inline RNCalendarEvents.m:681]: I am not familiar is iOS code, but this look like it's not that safe to return nil.
What is the goal of this? How does JS should behave with this change?

File: RNCalendarEvents.m [hunk 5]
```diff
@@ -664,7 +659,7 @@ - (NSDictionary *)serializeCalendarEvent:(EKEvent *)event
 
     [formedCalendarEvent setValue:[self availabilityStringMatchingConstant:event.availability] forKey:_availability];
     
-    if (event.structuredLocation && event.structuredLocation.title && event.structuredLocation.radius) {
+    if (event.structuredLocation) {
```
Comment (MoOx) [inline RNCalendarEvents.m:662]: Any reason you remove some safety tests?

File: RNCalendarEvents.m [hunk 6]
```diff
@@ -503,7 +498,7 @@ - (NSDictionary *)serializeCalendarEvent:(EKEvent *)event
 
     if (event.calendar) {
         [formedCalendarEvent setValue:@{
-                                        @"id": event.calendar.calendarIdentifier?event.calendar.calendarIdentifier: @"tempCalendar",
+                                        @"id": event.calendar.calendarIdentifier,
```
Comment (MoOx) [inline RNCalendarEvents.m:501]: Could you explain me this change please?

File: RNCalendarEvents.m [hunk 7]
```diff
@@ -814,27 +813,6 @@ - (NSDictionary *)serializeCalendarEvent:(EKEvent *)event
...
-        NSError *error = nil;
-
-        BOOL success = [self.eventStore removeCalendar:calendar commit:YES error:&error];
-        if (error) {
-            return reject(@"error", [error.userInfo valueForKey:@"NSLocalizedDescription"], nil);
-        }
-        return resolve(@(success));
-    });
-}
-
```
Comment (MoOx) [inline RNCalendarEvents.m:837]: ??

File: RNCalendarEvents.m [hunk 8]
```diff
@@ -864,7 +842,13 @@ - (NSDictionary *)serializeCalendarEvent:(EKEvent *)event
...
         NSArray *calendarEvents = [[strongSelf.eventStore eventsMatchingPredicate:predicate] sortedArrayUsingSelector:@selector(compareStartDateWithEvent:)];
         if (calendarEvents) {
-            resolve([strongSelf serializeCalendarEvents:calendarEvents]);
+           
+            NSDictionary * sCalendarEvent = [strongSelf serializeCalendarEvents:calendarEvents];
+            if(sCalendarEvent != nil){
+                    resolve(sCalendarEvent);
+            }else{
+                    resolve(@[]);
+            }
```
Comment (MoOx) [inline RNCalendarEvents.m:851]: Looks legit 👍

File: RNCalendarEvents.m [hunk 9]
```diff
@@ -886,7 +870,16 @@ - (NSDictionary *)serializeCalendarEvent:(EKEvent *)event
...
+            
+            
+            NSDictionary * sCalendarEvent = [strongSelf serializeCalendarEvent:calendarEvent];
+            if(sCalendarEvent != nil){
+                resolve(sCalendarEvent);
+                //[serializedCalendarEvents addObject:sCalendarEvent];
+            }else{
+                reject(@"error", @"error finding event", nil);
+            }
+
```
Comment (MoOx) [inline RNCalendarEvents.m:882]: How does the behave without this change?

**Comment by MoOx** [comment]
It seems you reverted https://github.com/wmcmahan/react-native-calendar-events/commit/4cb815b2b985d1e4f4fcf4397e2f317c7d9ca761 without paying attention :)

**Pull request closed without merging**